tests/conftest.py

```diff
@@ -0,0 +1,54 @@
+import flask
+import pytest
+from flask import template_rendered
+from main import create_app
+
+@pytest.fixture()
+def app():
+    app = create_app()
+    app.config.update({
+        "TESTING": True,
+    })
+
+    # other setup can go here
+
+    yield app
+
+    # clean up / reset resources here
+
+@pytest.fixture()
+def client(app):
+    return app.test_client()
+
+@pytest.fixture()
+def test_client():
+    # return TestApp(app)
+    flask_app = create_app()
+    flask_app.config.update({
+        'TESTING': True,
+    })
+ 
+    # Flask provides a way to test your application by exposing the Werkzeug test Client
+    # and handling the context locals for you.
+    testing_client = flask_app.test_client()
+ 
+    # Establish an application context before running the tests.
+    ctx = flask_app.app_context()
+    ctx.push()
+ 
+    yield testing_client  # this is where the testing happens!
+ 
+    ctx.pop()
+
+@pytest.fixture
+def captured_templates(app):
+    recorded = []
+
+    def record(sender, template, context, **extra):
+        recorded.append((template, context))
+
+    template_rendered.connect(record, app)
+    try:
+        yield recorded
+    finally:
+        template_rendered.disconnect(record, app)
```

Comment:
These are test fixtures. These run before the tests and provide important setup steps that are often reused.